flight/PiOS/Common/pios_dsm.c

```diff
@@ -229,7 +229,24 @@ int PIOS_DSM_UnrollChannels(struct pios_dsm_dev *dsm_dev)
 	// If no stream type has yet been detected, then try to probe for it
 	// this should only happen once per power cycle
 	if (dsm_dev->resolution == DSM_UNKNOWN) {
-		dsm_dev->resolution = PIOS_DSM_DetectResolution(state->received_data);
+
+		// check user settings to determine which resolution mode to use
+		uint8_t dsm_resolution_mode;
+		ManualControlSettingsDSMResolutionModeGet( &dsm_resolution_mode );
```

Comment:
We try to avoid the design pattern of pios drivers are reaching into UAVObjects. It would be better to move this into the part of the "userspace" code when the PIOS_DSM_Init is called and pass in a mode enum there.